### PR TITLE
Feature/687 mysql sink add support for dm-by-entity-type datamodel

### PR DESCRIPTION
**Pull request by pmo-sdr**
This PR adds support for a new entity-type datamodel for MySqlSink.

Issue #687

### CODE REVIEW

**Comment by pmo-sdr** [comment]
When it's true, table name is built with slash character. This results in table names like: 
dm-by-entity = `x002fxffff<entityId>xffff<entityType>`
dm-by-entity-type = `x002fxffff<entityType>`

We will have useless "x002fxffff" starting all table names.
Is it right?, looks strange for us. 

## DOC Reference:
https://github.com/telefonicaid/fiware-cygnus/blob/5361879d84b9e5bbb2da847ecef6d2fe3da97742/doc/cygnus-ngsi/flume_extensions_catalogue/ngsi_mysql_sink.md


The following table summarizes the table name composition (old encoding):

| FIWARE service path | `dm-by-service-path` | `dm-by-entity` | `dm-by-entity-type` |
|---|---|---|---|
| `/` | N/A | `<entityId>_<entityType>` | `<entityType>` |
| `/<svcPath>` | `<svcPath>` | `<svcPath>_<entityId>_<entityType>` | `<svcPath>_<entityType>` |

The following table summarizes the table name composition (new encoding):

| FIWARE service path | `dm-by-service-path` | `dm-by-entity` | `dm-by-entity-type` |
|---|---|---|---|
| `/` | `x002f` | `x002fxffff<entityId>xffff<entityType>` | `x002fxffff<entityType>` |
| `/<svcPath>` | `x002f<svcPath>` | `x002f<svcPath>xffff<entityId>xffff<entityType>` |`x002f<svcPath>xffff<entityType>` |

File: CHANGES_NEXT_RELEASE [hunk 1]
```diff
@@ -1 +1,2 @@
 - [cygnus-common] [NMI] Allows th Original values to be omitted in the configuration file. (#1466)
+- [MysqlSink] Adds support for dm-by-entity-type data model. (#687)
```
[the remark this answers is not in the document]
Comment (pmo-sdr) [reply CHANGES_NEXT_RELEASE]: 👍   1d8a40c

**Comment by fgalan** [comment]
I think that using a "/" as prefix is to mimic precisely the way in which the service path comes in the fiware-servicepath header. Looking to the past, maybe we could have assumed a "convenion" of removing that prefix in table names (as it is alwayw there) but it seems we didn't :). 

For the sake of homegeneity with other data models and backward compatibility I'd suggest continuing using this way.

**Comment by pmo-sdr** [comment]
Notice that MysqlSink removes slashes from table names when encoding = false.

**Comment by fgalan** [comment]
> Notice that MysqlSink already removes slashes from table names when encoding = false.

I'm afraid I don't know the answer for this behaviour sorry... but the conservative approach would be to keep it as it is. Do you agree? Is this PR ready to merge?

**Comment by pmo-sdr** [comment]
It looks strange for us having different behaviours depending on **encoding** value. But actually we are not using **encoding=true**, there may be a reason for leaving slash while enncoding, (We can't figure out what) and this is why we are asking. 

The change wouldn't be backward compatible, so we can leave this PR as is, and open a new Issue for this if you agree.

The PR is ready to merge if we are not going to change "slash behaviour" here.

**Comment by fgalan** [comment]
> The change wouldn't be backward compatible, so we can merge this PR as is, and open a new Issue for this if you agree.

I think is the best way to progress. Please, open the issue. Thanks!

**Pull request merged**